js/src/lib/adapters/datadog/mod.ts

```diff
@@ -134,6 +162,10 @@ export class DatadogAdapter implements Adapter {
     }
   }
 
+  public setMetadata(meta: DatadogMetadata): void {
```

Comment:
@bhelx what do you think about updating the interface to have this method instead of the placeholder I included here: https://github.com/dylibso/observe-sdk/blob/7637aa659d0dc865011bc4eaa14814a635009a49/js/src/lib/mod.ts#L65

I think if we use `setMetadata(meta: any): void` in the interface, it's ok to override w/ `DatadogMetadata` in the implementation.

Reply:
actually I think i'll just remove the addMetadata method from the collector interface... seeing your datadog adapter impl makes it more clear that this should be only on the adapter